x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -124,6 +124,8 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
 
                 if (filteredAlerts.length === 0) {
                   return { createdAlerts: [], errors: {}, alertsWereTruncated: false };
+                } else if (maxAlerts === 0) {
+                  return { createdAlerts: [], errors: {}, alertsWereTruncated: true };
```

Comment:
Short circuiting here if `maxAlerts` is passed in as zero. We're using this function's deduping ability to check that: if max signals was hit, are any remaining alerts actual alerts or are they duplicates? If there are remaining alerts in `filteredAlerts` at this point, we can assume they are actual alerts and don't need any additional function logic